### PR TITLE
WIP: debug pypy3 failures due to "Unroll calls to any #5062 (#5103)"

### PR DESCRIPTION
**Pull request by blueyed**
This reverts commit 2b9ca342809a4e05b4451a074d644c14acda844c, reversing
changes made to 0a57124063a4ed63f85fb8a39dc57f6d6fc5b132.

This is when pypy3 started to fail (https://travis-ci.org/pytest-dev/pytest/builds/537992776).

Ref: https://github.com/pytest-dev/pytest/issues/5317

/cc @Tadaboody

### CODE REVIEW

**Comment by nicoddemus** [comment]
How about if instead of reverting this completely, we don't try to unroll calls to `any` in pypy3 instead? It seems to me this would make it easier to fix the problem than trying to reintroduce the entire feature later.

**Comment by blueyed** [comment]
> How about if instead of reverting this completely

This is just for testing / investigating.  I do not want / plan to revert it actually (as-is).

**Comment by nicoddemus** [comment]
Oh OK!

**Comment by blueyed** [comment]
The build passed: https://travis-ci.org/pytest-dev/pytest/jobs/538946951
But IIRC it is rather flaky instead, due to something appearing on stderr.

**Comment by blueyed** [comment]
From the warnings summary:

> .tox/pypy3-xdist/site-packages/_pytest/fixtures.py:1270
  /home/travis/build/pytest-dev/pytest/.tox/pypy3-xdist/site-packages/_pytest/fixtures.py:1270: ResourceWarning: unclosed file <_io.FileIO fd=445 mode='rb' closefd=True>
    marker = getfixturemarker(obj)

**Comment by nicoddemus** [comment]
Thanks for the last commmit.

Argh this makes it harder to debug because I can't get `pypy3` to work on Windows. 😞

**Comment by blueyed** [comment]
Good first shot though I guess.
Might be worth to check if there are warnings also in pypy already, and/or if this is really just maybe flaky due to "output on stderr" (and happened before, but not that much).

**Comment by nicoddemus** [comment]
> Might be worth to check if there are warnings also in pypy already, and/or if this is really just maybe flaky due to "output on stderr" (and happened before, but not that much).

Hmmm interesting, we can try with `-s` next.

**Comment by blueyed** [comment]
> Hmmm interesting, we can try with -s next.

`-s` appears to be pretty broken in general already (started debugging / looking into it in https://github.com/pytest-dev/pytest/pull/5337 - which is against master)

**Comment by nicoddemus** [comment]
`-s` and xdist don't play nice together. We might try `pypy3` without xdist and only a single test file, to see if we can reproduce the issue with a smaller number of tests and shorter feedback. 🤔

**Comment by blueyed** [comment]
I could imagine that rewriting might happen in a place where PyPy does some resource handling itself (https://github.com/pytest-dev/pytest/pull/5103/files#diff-a54d1eaf9d064e9a7d25b1152476bb3dR967).

Bisecting/debugging this quickly is not fun, since there are warnings also on master already.
(Would be easier now with enforced `--lsof` being in place before already)

**Comment by blueyed** [comment]
Still appears to be flaky: https://travis-ci.org/pytest-dev/pytest/builds/539061916 ...
(via https://travis-ci.org/pytest-dev/pytest/branches)

**Comment by nicoddemus** [comment]
*sigh* now travis won't start

**Comment by nicoddemus** [comment]
It was because it had a merge conflict then.

File: .travis.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ before_script:
       export _PYTEST_TOX_EXTRA_DEP=coverage-enable-subprocess
     fi
 
-script: tox -- testing/acceptance_test.py
+script: tox -- testing/test_capture.py
```
Comment (blueyed) [inline .travis.yml]: In my manual testing it needs more, but `-m "not slow"` should do IIRC.

Comment (blueyed) [reply .travis.yml]: Should be rebased for that though I guess.

Comment (nicoddemus) [reply .travis.yml]: Trying `testing/python` now

But what do you suggest, `testing -m "not slow"`?

**Comment by blueyed** [comment]
btw, just out of interest: could you use pypy3 in Docker maybe?
Also the nt._getfinalpathname should get reported/fixed probably.

File: .travis.yml (hunk 1 above)
Comment (blueyed) [reply .travis.yml]: Yes.. just `-m "not slow"` basically.

**Comment by nicoddemus** [comment]
> btw, just out of interest: could you use pypy3 in Docker maybe?

I have one or two VMs laying around, I guess there's no other way but try this.

> Also the nt._getfinalpathname should get reported/fixed probably.

Definitely, didn't do that yesterday as it was late and I was about to clock out. Will do it later.

**Comment by nicoddemus** [comment]
Spent the hour or so investigating this. Some observations:

1. I get warnings by just running `tox -e pypy3 -- testing/acceptance_test.py`.
2. If I add `--runpytest=subprocess`, I don't see any warnings. 🤔 

I will clock out now, might give this another try tomorrow. 

But I'm not confident we can find the problem anytime soon. The question is, should this block the 4.6 release? IMHO no; while it is unfortunate that we are getting more of those warnings now, it seems they affect mostly pytest's own test suite, so this is something we can deal later.

**Comment by blueyed** [comment]
> should this block the 4.6 release?

No.
We should either allow failures there, or just remove it temporarily.

**Comment by nicoddemus** [comment]
I agree with allowing failures. I believe you had a PR with that approach right? Could you resurrect it please? 😁

**Comment by blueyed** [comment]
Closing for now.

**Pull request closed without merging**